python_modules/dagster/dagster/_core/definitions/asset_selection.py

```diff
@@ -436,11 +451,17 @@ def resolve_checks_inner(self, asset_graph: InternalAssetGraph) -> AbstractSet[A
             asset_graph
         )
 
+    def to_serializable_asset_selection(self, asset_graph: AssetGraph) -> "AssetSelection":
+        return self._replace(
```

Comment:
Not a super strongly-held opinion so feel free to ignore, but it's a bit odd to use `_replace` here when we're not preserving any of the original properties.

In the extremely unlikely event that we add additional fields to these asset selections or change their names, it might be more type system-friendly to just create a new `AndAssetSelection` instance

Reply:
My thinking here was to try to defend against copy/paste errors like the one here: https://github.com/dagster-io/dagster/pull/18777.

Reply:
Fair enough 👍